src/common/serac_types.hpp

```diff
@@ -203,6 +204,68 @@ struct BoundaryCondition {
   std::unique_ptr<mfem::HypreParMatrix> eliminated_matrix_entries;
 };
 
+/**
+ * Wraps a (currently iterative) system solver and handles the configuration of linear
+ * or nonlinear solvers
+ */
+class SystemSolver : public mfem::Solver {
```

Comment:
`SystemSolver` is a bit overloaded. Maybe `LinearAlgegbraSolver` to denote that these are numerics operators?

Reply:
Although the nonlinear one is here too. Hmmmm.

Reply:
`AlgebraicSolver`?